src/Streamly/Internal/Data/Stream/Async/Channel.hs

```diff
@@ -35,19 +34,25 @@ module Streamly.Internal.Data.Stream.Async.Channel
     -- * Stream operations
     , evalWithK
     , appendWithK
-    , concatMapWithK
+    , interleaveWithK
+    , concatMapWithK -- appendMap
+    , concatMapInterleaveWithK -- interleaveMap
```

Comment:
If you're exporting `withChannelK`, Also export `withInterleaveChannelK`? Keeping the corresponding interleave function will help make the distinction clearer and in a sense describe if it's a general API or async strategy specific..

Reply:
Also, `evalWithInterleaveK`.

Reply:
append or interleave does not apply to eval as it is a single stream. In fact, eval should only exist in the Stream.Parallel module.